### ISSUE TITLE
shada error: invalid line number after `nvim -u NORC +term`

### ISSUE THREAD

**Issue by JulioJu** [neovim/neovim#7677]
- `nvim --version`: NVIM v0.2.2
- Vim (version: ) behaves differently? irrelevant
- Operating system/version: Arch Linux
- Terminal name/version: GNOME Terminal 3.26.2 Using VTE version 0.50.2 +GNUTLS or TTY
- `$TERM`: xterm-256color or linux

### Steps to reproduce using `nvim -u NORC`
With:
- shell: zsh with or without `.zshrc` or sh
- ~/.local/share/nvim/shada/main.shada existing or not exiting.

```
nvim -u NORC +term
:q
nvim

```
### Actual behaviour
Cause output error:
```
E575: Error while reading ShaDa file: mark entry at position [number] has invalid line number
Press ENTER or type command to continue
```
### Expected behaviour
This error should not be displayed

\[note: should not be the same as https://github.com/neovim/neovim/issues/4100 \]
\[note: this error not occur if in the vimrc there is `autocmd BufWinEnter,WinEnter * startinsert` as I've explained in https://github.com/neovim/neovim/issues/7678 \]

**Comment by justinmk** [comment]
> [note: this error not occur if in the vimrc there is autocmd BufWinEnter,WinEnter * startinsert as I've explained in #7678 ]

Does https://github.com/neovim/neovim/pull/7431 fix that by chance?

**Comment by JulioJu** [comment]
@justinmk not, does not seems to fix it.
This issue is a regression between neovim 0.2.0 and neovim 0.2.1.
There was not problem with that in neovim 0.2.0.
Contrary to #7678, the bisject isn't https://github.com/neovim/neovim/commit/7d183e89f7c055c8f5217e91294123a5aa0a5e2d. Its parent https://github.com/neovim/neovim/commit/8d1ccb606d3806dcf9c4bcfdb0446d8139c0e765 has already this issue.
I don't know how to bisject. @nateozem, could you also bisject  or give me the script please ?

**Comment by justinmk** [comment]
Can reproduce starting with a new shada file.

    nvim -u NORC +term -i foo

**Comment by JulioJu** [comment]
Hi
I don't know if it's useful, but I've searched the problematic commit. The author is @ckelsel. 
```
$ git bisect bad
348229b1d32010292013eee415c28926e2b7009b is the first bad commit
commit 348229b1d32010292013eee415c28926e2b7009b
```
```
$ git log --pretty=fuller
commit 348229b1d32010292013eee415c28926e2b7009b (HEAD, refs/bisect/bad)
Author:     KunMing Xie <qqzz014@gmail.com>
AuthorDate: Wed Aug 2 16:22:04 2017 +0800
Commit:     Justin M. Keyes <justinkz@gmail.com>
CommitDate: Fri Aug 4 22:25:39 2017 +0200

    vim-patch:8.0.0133 (#7107)
    
    Problem:    "2;'(" causes ml_get errors in an empty buffer.  (Dominique Pelle)
    Solution:   Check the cursor line earlier.
    
    https://github.com/vim/vim/commit/fe38b494fff56cd9b2fcaeef26a8fd7b6557d69c
```
The corresponding merge is #7107 and the link to the commit is neovim/neovim@348229b1.

**Comment by justinmk** [comment]
That's definitely useful!

**Comment by justinmk** [comment]
Fixed by 6b45dbca0429
